File: demo_core_av/app/src/main/java/com/bandyer/demo_core_av/design/adapter_items/StringItem.java
/*
 * Copyright (C) 2018 Bandyer S.r.l. All Rights Reserved.
 * See LICENSE.txt for licensing information
 */

package com.bandyer.demo_core_av.design.adapter_items;

import androidx.annotation.NonNull;
import android.view.View;
import android.widget.TextView;

import com.bandyer.demo_core_av.R;
import com.bandyer.demo_core_av.room.adapter_items.ButterKnifeViewHolder;
import com.mikepenz.fastadapter.items.AbstractItem;

import java.util.List;

import butterknife.BindView;


/**
 * @author kristiyan
 **/

public class StringItem extends AbstractItem<StringItem, StringItem.Holder> {

    private String text;

    public StringItem(String text) {
        this.text = text;
    }

    @NonNull
    @Override
    public Holder getViewHolder(@NonNull View v) {
        return new StringItem.Holder(v);
    }

    @Override
    public int getType() {
        return R.id.stream_item;
    }

    @Override
    public int getLayoutRes() {
        return R.layout.string_item;
    }

    public static class Holder extends ButterKnifeViewHolder<StringItem> {

        @BindView(R.id.text)
        TextView textView;

        Holder(View view) {
            super(view);
        }

        @Override
        public void bindView(@NonNull StringItem stringItem, @NonNull List<Object> payloads) {
            textView.setText(stringItem.text);
        }

        @Override
        public void unbindView(@NonNull StringItem item) {
        }
    }
}

File: demo_core_av/app/src/main/java/com/bandyer/demo_core_av/design/adapter_items/StringItem.kt
/*
 * Copyright (C) 2018 Bandyer S.r.l. All Rights Reserved.
 * See LICENSE.txt for licensing information
 */
package com.bandyer.demo_core_av.design.adapter_items

import android.view.View
import com.bandyer.demo_core_av.R
import com.mikepenz.fastadapter.FastAdapter
import com.mikepenz.fastadapter.items.AbstractItem
import kotlinx.android.extensions.LayoutContainer
import kotlinx.android.synthetic.main.string_item.view.*

/**
 * @author kristiyan
 */
class StringItem(val text: String?) : AbstractItem<StringItem?, Holder>() {
    override fun getLayoutRes(): Int = R.layout.string_item
    override fun getViewHolder(v: View): Holder = Holder(v)
    override fun getType(): Int = R.id.stream_item
}

class Holder internal constructor(override val containerView: View) : FastAdapter.ViewHolder<StringItem>(containerView), LayoutContainer {

    override fun bindView(stringItem: StringItem, payloads: List<Any>) {
        containerView.textView!!.text = stringItem.text
    }

    override fun unbindView(item: StringItem) = Unit
}